code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -325,14 +326,17 @@ var/global/list/damage_icon_parts = list()
 //		else
 			//warning("Invalid my_appearance.f_style for [species.name]: [my_appearance.f_style]")
 
-	if(my_appearance.h_style && !(check_hidden_flags(get_clothing_items(),HIDEHEADHAIR)))
```

Comment:
Why remove this?

Reply:
I don't exactly remember my thought process at the time I made the OG change but I can assure there's a reason I didn't use `check_hidden_flags`.
Two most probable answers are that I either was too lazy to decipher the proc and make it work with my system, or that it conflicted with other behaviour.

I think the three-tiers system (doesn't hide hair - caps, dermal -, hide hair partially - sec helmets, riot helmets -, and fully hide hair - spacesuit helmets, ushankas, etc) is simple enough to understand without being that much of an overhead.

Reply:
My thought is this; if you want some helmets, like your sec helmets and hos caps, to only partially hide hair, why not make a body obscure flag for that rather than add this new flag and ignoring this proc entirely?

The only reason your method works currently is due to the oversight that was not using this function to check headgear

Reply:
Same result, different jargon

Reply:
>same result, different jargon

Not really